### PR TITLE
Fix logging demo test

### PR DESCRIPTION
**Pull request by hidmic**
By relocating `output_filter` assignment. Running CI for Linux only as changes are cross-platform.

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/8412/)

### CODE REVIEW

**Comment by dirk-thomas** [comment]
Can you describe why `setUpClass` isn't sufficient? Is it not being called?

Also retriggered build passed: [Build Status](http://ci.ros2.org/job/ci_linux/8413/)

**Comment by hidmic** [comment]
> Can you describe why `setUpClass` isn't sufficient? Is it not being called?

At first it wasn't obvious to me either.

But deep down, it's because of [this](https://github.com/ros2/launch/pull/279/files#diff-c9721705993412e7b327ce2112fa3469R187-R191). That patch essentially duplicates test case classes through inheritance to prevent overlap of arguments bound to class methods for launch tests that use parameterization -- there's only one class definition and N different set of arguments to be bound. Now, because of that, when `setUpClass()` gets called `cls` becomes that subclass and later attempts to refer to any of the set attributes via the base class fail.

I'm not particularly happy with either change, but I deem this one less harmful. The true problem is that `launch_testing` parameterization is not correct in this regard. `pytest` solves this problem in completely different way, by providing "class"-scoped fixtures (see [here](https://stackoverflow.com/questions/19736363/pytest-how-to-pass-a-class-parameter-to-setup-class)).

**Comment by hidmic** [comment]
Or, alternatively, we can forbid `launch_testing` to bind `setUpClass()` arguments and change CLI tests accordingly. It doesn't look pretty, that's why it wasn't my first choice.

File: logging_demo/test/test_logging_demo.py.in [hunk 1]
```diff
@@ -38,12 +38,9 @@ def generate_test_description(ready_fn):
 
 
 class TestLoggingDemo(unittest.TestCase):
-
```
Comment (dirk-thomas) [inline logging_demo/test/test_logging_demo.py.in]: You might want to keep this empty line?

Comment (hidmic) [reply logging_demo/test/test_logging_demo.py.in]: Done in https://github.com/ros2/demos/pull/400/commits/9056e2d1932c10558b5e37e6ad2b60b96d1dfab8

**Pull request merged**